proposed/http-middleware/middleware-meta.md

```diff
@@ -270,75 +286,48 @@ additional information._
 
 [promises]: https://promisesaplus.com/
 
-### 5.2 Delegate Design
-
-The `DelegateInterface` defines a single method that accepts a request and
-returns a response. The delegate interface must be implemented by any middleware
-dispatcher that uses middleware implementing `MiddlewareInterface`.
-
-#### Why the term "delegate"?
-
-The term "delegate" means something designated to act for or represent another.
-In terms of middleware design, a delegate is called upon by middleware when the
-middleware is unable to process the request itself; the delegate then processes
-the request _for the original middleware_ in order to return a response.
-
-#### Why isn't the delegate a `callable`?
+### 5.2 Handler Design
 
-Using an interface type hint improves runtime safety and IDE support.
+The `HandlerInterface` defines a single method that accepts a request and
+MUST return a response. The handler MAY delegate to another handler.
 
-_See "discussion of FrameInterface" in [relevant links](#8-relevant-links) for
-additional information._
-
-#### Why doesn't delegate use `__invoke`?
-
-Classes that define `__invoke` can be more loosely type hinted as
-`callable`, which results in less strict typing. This is generally undesirable,
-especially when the method uses strict typing.
-
-#### Why does the delegate conflict with middleware?
-
-Both the middleware and delegate interface define a `process` method to
-discourage misuse of middleware as delegates.
-
-The implementation of the delegate should be defined within middleware
-dispatching systems.
+#### Why is a server request required?
 
-#### Why not the term `$next`?
+To make it clear that the handler can only be used in a server side context.
+In an client side context a [promise][promises] would likely be returned instead
+of a response.
 
-Several existing middleware libraries use the term `$next` instead of
-`$delegate`. `$next` implies an action: "Next, please!" As such, these libraries
-define `$next` as a `callable`, which we note was undesirable for purposes of
-this specification in the previous section.
+#### Why the term "handler"?
 
-Additionally, since we are defining an object, we chose to use a noun instead of
-a verb to name the interface.
+The term "handler" means something designated to manage or control. In terms of
+request processing, a handler is the point where the request must be acted upon
+to create a response.
 
-Further, we did not choose the term `next` for the action delegates invoke, as
-that verb implies a queue or stack. The delegate is not required to implement
-either pattern internally in order to do its work; its only job is to _process_
-the request to return a response.
+As opposed to the term "delegate", which was used in a previous version of this
+specification, the internal behavior of the this interface is not specified.
+As long as the handler ultimately produces a response, it is considered valid.
```

Comment:
@weierophinney I think this correctly addresses the historical reasons for `handler` as opposed to `delegate`. Thoughts?